src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -150,15 +150,20 @@ public override PSThreadOptions ThreadOptions
                     {
                         if (this.RunspaceStateInfo.State != RunspaceState.BeforeOpen)
                         {
-#if CORECLR                 // No ApartmentState.STA Support In CoreCLR
-                            bool allowed = value == PSThreadOptions.ReuseThread;
-#else
-                            // if the runspace is already opened we only allow changing the options if
-                            // the apartment state is MTA and the new value is ReuseThread
-                            bool allowed = (this.ApartmentState == ApartmentState.MTA || this.ApartmentState == ApartmentState.Unknown) // Unknown is the same as MTA
-                                           &&
-                                           value == PSThreadOptions.ReuseThread;
-#endif
+                            bool allowed = false;
+
+                            if (Platform.IsNanoServer || Platform.IsIoT)
+                            {
```

Comment:
Maybe we should also fall in this code path when on Linux and macOS? That's the case before this change ...